script/main.js

```diff
@@ -75,30 +71,35 @@ async function showNotebook(){
 
 }
 async function deleteNote(){
-
-    const response = await fetch(`${APIUrl}/delete`,{
-        method: 'DELETE',
-        headers: {
-            'Content-type':'application/json'
-        },
-        body: JSON.stringify({id: valueInput})
-    })
-
-    if (!response.ok){
-        throw new Error('There is a problem.');
-    }
-    const result = await response.json();
-    console.log(`The result is: ${result}`)
+    const deleteButtons = document.querySelectorAll('.fa-trash-alt');
+
+    deleteButtons.forEach(button => {
+        button.addEventListener('click', async function (event) {
+            event.stopPropagation();
+            const noteContainer = this.closest('.notebook-single-container');
+            const noteId = noteContainer.firstElementChild.getAttribute('data-id');
+
+            const response = await fetch(`${APIUrl}/delete`,{
```

Comment:
Because you use await, its best practice to use try-catch with it. for better error handling. 
You can then trigger an alert to show the user that something went wrong. and also do a conspole.error of the error message you get from the backend to show the things that went wrong.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch